core/src/optimistic_confirmation_verifier.rs

```diff
@@ -175,7 +191,8 @@ mod test {
         {
             let blockstore = Blockstore::open(&blockstore_path).unwrap();
             let optimistic_slots = vec![(1, bad_bank_hash), (3, Hash::default())];
-            optimistic_confirmation_verifier.add_new_optimistic_confirmed_slots(optimistic_slots);
+            optimistic_confirmation_verifier
+                .add_new_optimistic_confirmed_slots(optimistic_slots, None);
```

Comment:
This test has a blockstore instance already.  Let's pass-in the blockstore and verify the written optimistic_slots at the end of the test as well?

Reply:
updated tests